core/store/src/trie/mod.rs

```diff
@@ -454,17 +508,18 @@ impl TrieChanges {
     }
     pub fn insertions_into(
         &self,
-        trie: Arc<Trie>,
+        trie: Arc<ShardTries>,
```

Comment:
Why cannot we keep it simple single Trie? I mean, what is the reason to pass a list of all the ShardTries and shard_id index all the way down?

This API is confusing now as it is hard to reason whether one shard can influence other shards.

Reply:
Here the reason is that someone needs to pass `ShardTries` to `StoreUpdate`.

Reply:
Can we decouple that? Create something like TrieUpdate and operate on that level?

Reply:
How about we move storage mutation methods to `ShardTries`: `ShardTries::apply_insertions(ShardId, &TrieChanges, &mut StoreUpdate)`?

Reply:
@mikhailOK I am not sure I exactly follow the suggestion, so let me try to clarify:

Having the change you proposed we will keep the Trie operations as they used to be (operating on a single Trie and returning TrieChanges), and then, one level up, we apply the TrieChanges into StoreUpdate. This sounds reasonable to me :+1:

Reply:
moved the code out of TrieChanges, now only `WrappedTrieChanges` and `ShardTries` are aware of shard_id